src/ctrace/src/ctrace_config.erl

```diff
@@ -98,17 +98,37 @@ maybe_start_main_tracer(TracerId) ->
 
 
 start_main_tracer(TracerId) ->
-    Sampler = passage_sampler_all:new(),
-    Options = [
-        {thrift_format,
-            list_to_atom(config:get("tracing", "thrift_format", "compact"))},
-        {agent_host, config:get("tracing", "agent_host", "127.0.0.1")},
-        {agent_port, config:get_integer("tracing", "agent_port", 6831)},
-        {default_service_name,
-            list_to_atom(config:get("tracing", "app_name", "couchdb"))}
-    ],
-    ok = jaeger_passage:start_tracer(TracerId, Sampler, Options).
+    MaxQueueLen = config:get_integer("tracing", "max_queue_len", 1024),
+    Sampler = jaeger_passage_sampler_queue_limit:new(
+        passage_sampler_all:new(), TracerId, MaxQueueLen),
+    ServiceName = list_to_atom(config:get("tracing", "app_name", "couchdb")),
+
+    case config:get("tracing", "protocol", "udp") of
+        "udp" ->
+            Options = [
+                {thrift_format, list_to_atom(
+                    config:get("tracing", "thrift_format", "compact"))},
+                {agent_host,
+                    config:get("tracing", "agent_host", "127.0.0.1")},
+                {agent_port,
+                    config:get_integer("tracing", "agent_port", 6831)},
+                {protocol, udp},
+                {default_service_name, ServiceName}
+           ],
+           jaeger_passage:start_tracer(TracerId, Sampler, Options);
```

Comment:
Minor nit, but the previous version of this function matched `ok` against `jaeger_passage:start_tracer(...)`, whereas this version does not. Was that an intentional behavior change?

Reply:
Fixed in https://github.com/apache/couchdb/pull/2494/commits/a5445d7d949abe0f5c0df2584b0ef5cbc6964ae6